src/overlays/actors/ovl_Bg_Breakwall/z_bg_breakwall.c

```diff
@@ -53,21 +66,268 @@ static ColliderQuadInit D_80870760 = {
+            for (k = 3; k >= 0; k--) {
+                Matrix_MultVec3f(&actorPosList[j][k], &actorPos);
+                actor =
+                    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_A_OBJ, Rand_CenteredFloat(20.0f) + actorPos.x,
+                                Rand_CenteredFloat(20.0f) + actorPos.y, Rand_CenteredFloat(20.0f) + actorPos.z,
+                                actorRotList[k].x, actorRotList[k].y + angle1, actorRotList[k].z, 0x000B);
+
+                if ((j & 1) == 0) {
+                    func_80033480(globalCtx, &actorPos, velocity * 200.0f, 1, 650, 150, 1);
+                }
+
+                if (actor != NULL) {
+                    actor->speedXZ = Rand_ZeroOne() + (accel * 0.6f);
+                    actor->velocity.y = Rand_ZeroOne() + (accel * 0.6f);
+                    actor->world.rot.y += (s16)((Rand_ZeroOne() - 0.5f) * 3000.0f);
+                    actor->world.rot.x = (s16)(Rand_ZeroOne() * 3500.0f) + 2000;
+                    actor->world.rot.z = (s16)(Rand_ZeroOne() * 3500.0f) + 2000;
+                    actor->parent = &this->dyna.actor;
+                    actor->scale.x = actorScaleList[k].x + Rand_CenteredFloat(0.001f);
+                    actor->scale.y = actorScaleList[k].y + Rand_CenteredFloat(0.001f);
+                    actor->scale.z = actorScaleList[k].z + Rand_CenteredFloat(0.001f);
+                }
+            }
+        }
+    }
+
+    return actor;
+}
+
+/**
+ * Sets up the collision model as well is the object dependency and action function to use.
+ */
+void BgBreakwall_SetupWait(BgBreakwall* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->bankIndex)) {
+        CollisionHeader* colHeader = NULL;
+        s32 wallType = ((this->dyna.actor.params >> 13) & 3) & 0xFF;
+
+        this->dyna.actor.objBankIndex = this->bankIndex;
+        Actor_SetObjectDependency(globalCtx, &this->dyna.actor);
+        this->dyna.actor.flags &= ~0x10;
+        this->dyna.actor.draw = BgBreakwall_Draw;
+        CollisionHeader_GetVirtual(sBombableWallInfo[wallType].colHeader, &colHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+
+        if (wallType == BWALL_KD_LAVA_COVER) {
+            BgBreakwall_SetupAction(this, BgBreakwall_LavaCoverMove);
+        } else {
+            BgBreakwall_SetupAction(this, BgBreakwall_Wait);
+        }
+    }
+}
+
+/**
+ * Checks for an explosion using quad collision. If the wall or floor is exploded then it will spawn fragments and
+ * despawn itself.
+ */
+void BgBreakwall_Wait(BgBreakwall* this, GlobalContext* globalCtx) {
+    if (this->collider.base.acFlags & 2) {
+        Vec3f effectPos;
+        s32 wallType = ((this->dyna.actor.params >> 13) & 3) & 0xFF;
+
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+
+        effectPos.x = 0.0f;
+        effectPos.z = 0.0f;
+        effectPos.y = 0.0f;
```

Comment:
possible chained assign? probably in xyz order.

Reply:
xyz order did not work, but i didn't try xz at once and then y yet which may work